cli/command/system/completion.go

```diff
@@ -0,0 +1,99 @@
+package system
+
+import (
+	"strings"
+
+	"github.com/docker/cli/cli/command/completion"
+	"github.com/spf13/cobra"
+)
+
+var (
+	eventFilters = []string{"container", "daemon", "event", "image", "label", "network", "node", "scope", "type", "volume"}
+	eventNames   = []string{
+		"attach",
+		"commit",
+		"connect",
+		"copy",
+		"create",
+		"delete",
+		"destroy",
+		"detach",
+		"die",
+		"disable",
+		"disconnect",
+		"enable",
+		"exec_create",
+		"exec_detach",
+		"exec_die",
+		"exec_start",
+		"export",
+		"health_status",
+		"import",
+		"install",
+		"kill",
+		"load",
+		"mount",
+		"oom",
+		"pause",
+		"pull",
+		"push",
+		"reload",
+		"remove",
+		"rename",
+		"resize",
+		"restart",
+		"save",
+		"start",
+		"stop",
+		"tag",
+		"top",
+		"unmount",
+		"unpause",
+		"untag",
+		"update",
+	}
+)
+var eventTypes = []string{"config", "container", "daemon", "image", "network", "node", "plugin", "secret", "service", "volume"}
+
+func completeFilters(dockerCLI completion.APIClientProvider) completion.ValidArgsFn {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		if strings.HasPrefix(toComplete, "container=") {
+			// the pure container name list should be pulled out from ContainerNames.
+			names, _ := completion.ContainerNames(dockerCLI, true)(cmd, args, toComplete)
+			return prefixWith("container=", names), cobra.ShellCompDirectiveDefault
+		}
+		if strings.HasPrefix(toComplete, "event=") {
+			return prefixWith("event=", eventNames), cobra.ShellCompDirectiveDefault
+		}
+		if strings.HasPrefix(toComplete, "label=") {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		if strings.HasPrefix(toComplete, "network=") {
+			names, err := completion.NetworkNames(dockerCLI, cmd)
```

Comment:
I'm slightly on the fence on the `completion` package now having both `completion.NetworkNames`  and `completion.CompleteNetworks`.  Or at least, I"m looking if we can reduce exported functions, unless needed.

Looking at the code in `NetworkNames`, it's just a few lines, and fairly straightforward. I'd be fine with copying that code here as a un-exported function (for now). It may be good to include a reference to the "original" one to correlate them, e.g.;


```go
// networkNames contacts the API to get a list of networks.
//
// This function is based on [completion.NetworkNames].
func NetworkNames(dockerCLI APIClientProvider, cmd *cobra.Command) ([]string, error) {
	summaries, err := dockerCLI.Client().NetworkList(cmd.Context(), network.ListOptions{})
	if err != nil {
		return nil, err
	}
	var names []string
	for _, nws := range summaries {
		names = append(names, nws.Name)
	}
	return names, nil
}
```


That would still allow us to change our mind in future of course!

Reply:
> That would still allow us to change our mind in future of course!

Expanding a bit on that (just to echo my train of thought ❤️); with cli-plugins, like compose, buildx, and others, we may need to think what to put "where" at some point. We didn't discuss that in-depth when we added the initial implementation of cobra completion, but sometimes it can be "complicated" to have the code live together with other code.

So, possibly at some point we may need to re-organise some of the functions to be exported (allowing them to be  consumed by compose and buildx, etc), but without that meaning they come with a large number of dependencies.

Reply:
> I'm slightly on the fence on the completion package now having both completion.NetworkNames and completion.CompleteNetworks.

I agree this feels a bit overkill for the desired effect.

> Looking at the code in `NetworkNames`, it's just a few lines, and fairly straightforward. I'd be fine with copying that code here as a un-exported function (for now).

For `NetworkNames` this would be a good solution. But for `ContainerNames`, this would mean a lot of duplicated business logic.

What about keeping the
```go
    names, _ := completion.NetworkNames(dockerCLI)(cmd, args, toComplete)
```
and
```go
    names, _ := completion.ContainerNames(dockerCLI, true)(cmd, args, toComplete)
```
hacks instead?

Reply:
> What about keeping the hacky version from before the refactoring:

Oh, I think that's definitely acceptable for now. Once we figure out alternatives, we can easily swap out those lines.

Reply:
I reverted to the previous version and added error handling for network and container completion to avoid filenames being completed in case of an error in the api calls.